rust_src/src/charset.rs

```diff
@@ -18,6 +21,20 @@ impl LispObject {
     }
 }
 
+/// Internal use only.
+/// Clear temporary charset mapping tables.
+/// It should be called only from temacs invoked for dumping.
+#[lisp_fn]
+pub fn clear_charset_maps() {
+    unsafe {
+        *temp_charset_work = Default::default();
```

Comment:
Is this OK, or should I assign `ptr::null_mut()` here instead? The C code freed the memory explicitly and assigned NULL. Would assigning `ptr::null_mut()` in Rust automatically free the memory?

Reply:
AFAIK C still handles memory, so I guess you have to free it.

Reply:
I apologize for the silly question, but I want to be sure I'm doing the right thing.

From what I've read Rust automatically frees memory once the values in question fall out of scope, but this function has several things going on with it that make me less sure this would happen here (this is a raw mutable pointer to a struct shared with C code via FFI).

If I replace the assignment above with `temp_charset_work = ptr:null_mut()`, will Rust still free the memory at the end of this function, or do I have to do something else to ensure the memory is freed?

Reply:
Don't worry, I still don't understand most of this stuff after almost 2 years =)
But I think you have to do it the same way. Rust doesn't free globals from C.